sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/display/DisplayDataTest.java

```diff
@@ -327,6 +328,81 @@ public void populateDisplayData(Builder builder) {
     DisplayData.from(component); // should not throw
   }
 
+  @Test
+  public void testRootPath() {
+    DisplayData.Path root = DisplayData.Path.root();
+    assertThat(root.getComponents(), Matchers.empty());
+  }
+
+  @Test
+  public void testExtendPath() {
+    DisplayData.Path a = DisplayData.Path.root().extend("a");
+    assertThat(a.getComponents(), hasItems("a"));
+
+    DisplayData.Path b = a.extend("b");
+    assertThat(b.getComponents(), hasItems("a", "b"));
+  }
+
+  @Test
+  public void testExtendNullPathValidation() {
+    DisplayData.Path root = DisplayData.Path.root();
+    thrown.expect(NullPointerException.class);
+    root.extend(null);
+  }
+
+  @Test
+  public void testExtendEmptyPathValidation() {
+    DisplayData.Path root = DisplayData.Path.root();
+    thrown.expect(IllegalArgumentException.class);
+    root.extend("");
+  }
+
+  @Test
+  public void testAbsolute() {
+    DisplayData.Path path = DisplayData.Path.absolute("a", "b", "c");
+    assertThat(path.getComponents(), hasItems("a", "b", "c"));
+  }
+
+  @Test
+  public void testAbsoluteValidation() {
+    try {
```

Comment:
Because it means I need to split them into separate test cases.. which is probably the right thing to do anyway. :) Thanks for pointing it out.